Sources/MongoSwift/MongoClient.swift

```diff
@@ -4,17 +4,12 @@ import NIOConcurrencyHelpers
 
 /// Options to use when creating a `MongoClient`.
 public struct ClientOptions: CodingStrategyProvider, Decodable {
-    /**
-     * Indicates whether this client should publish command monitoring events. If true, the following event types will
-     * be published, under the listed names (which are defined as static properties of `Notification.Name`):
-     * - `CommandStartedEvent`: `.commandStarted`
-     * - `CommandSucceededEvent`: `.commandSucceeded`
-     * - `CommandFailedEvent`: `.commandFailed`
-     */
-    public var commandMonitoring: Bool = false
-
     // swiftlint:disable redundant_optional_initialization
 
+    /// Indicates whether this client should publish command monitoring events. If non-nil, `CommandEvent`s will be
+    /// published to the provided handler.
+    public var commandEventHandler: CommandEventHandler? = nil
```

Comment:
Instead of a boolean flag, monitoring will simply be enabled when a monitor is present.